src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/Token.java

```diff
@@ -0,0 +1,188 @@
+package com.atlassian.bitbucket.jenkins.internal.applink.oauth;
+
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.consumer.Consumer;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Base type for OAuth tokens.  This type should never be used directly.  Instead, use the more specific
```

Comment:
What benefit does this split give us as opposed to just one class (either called `Token` or `ServiceProviderToken`)?

Reply:
I was also thinking about this. However I believe with upcoming work like trusted builds, we will need to be OAuth consumer for Bitbucket Server.  `Token` class will have a simple implementation for the consumer side of things.